File: 2352-equal-row-and-column-pairs/2352-equal-row-and-column-pairs.cs
public class Solution {
    public int EqualPairs(int[][] grid) {
        var dic = new Dictionary<string, int>();
        int ret = 0;
        
        foreach(int[] g in grid)
        {
            var arrStr = string.Join(" ", g);
            if(!dic.TryAdd(arrStr, 1)) dic[arrStr]++;
            //dic[arrStr] = dic.GetValueOrDefault(arrStr, 0)+1;            
            //if(!dic.ContainsKey(arrStr)) dic.Add(arrSrt, 0);                
            //dic[arrStr]++;            
        }
        
        for(int i=0; i < grid.Length; i++)
        {
            int[] colArr = new int[grid.Length];
            for(int j=0; j < grid.Length; j++)
            {
                colArr[j] = grid[j][i];
            }
            ret += dic.GetValueOrDefault(string.Join(" ", colArr), 0);
        }        
        return ret;        
    }
}

File: 2352-equal-row-and-column-pairs/2352-equal-row-and-column-pairs.ts
function equalPairs(grid: number[][]): number {
    const map: Map<string, number> = new Map();
    let ret: number = 0;
    
    grid.forEach((item)=>{
        const arrStr:string = item.join(' ');
        map[arrStr] = (map[arrStr] || 0) +1;
    });

    for(let i=0; i< grid.length; i++){
        const arr = [];
        for(let j=0; j < grid.length; j++){
            arr.push(grid[j][i]);
        }

        ret+= (map[arr.join(' ')] || 0);
    }
    
    return ret;
};